ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/MarkerDemoActivity.kt

```diff
@@ -0,0 +1,494 @@
+                )
+
+        )
+
+        // add 4 markers on top of each other in Darwin with varying z-indexes
+        (0 until 4).map {
+            placeDetailsMap.put(
+                "DARWIN ${it + 1}", PlaceDetails(
+                    position = places.getValue("DARWIN"),
+                    title = "Darwin Marker ${it + 1}",
+                    snippet = "z-index initially ${it + 1}",
+                    zIndex = it.toFloat()
+                )
+            )
+        }
+
+        // place markers for each of the defined locations
+        placeDetailsMap.keys.map {
+            with(placeDetailsMap.getValue(it)) {
+                map.addMarker(MarkerOptions()
+                        .position(position)
+                        .title(title)
+                        .snippet(snippet)
+                        .icon(icon)
+                        .infoWindowAnchor(infoWindowAnchorX, infoWindowAnchorY)
+                        .draggable(draggable)
+                        .zIndex(zIndex))
+
+            }
+        }
+
+        // Creates a marker rainbow demonstrating how to create default marker icons of different
+        // hues (colors).
+        val numMarkersInRainbow = 12
+        (0 until numMarkersInRainbow).mapTo(markerRainbow) {
+            map.addMarker(MarkerOptions().apply{
+                position(LatLng(
+                        -30 + 10 * Math.sin(it * Math.PI / (numMarkersInRainbow - 1)),
+                        135 - 10 * Math.cos(it * Math.PI / (numMarkersInRainbow - 1))))
+                title("Marker $it")
+                icon(BitmapDescriptorFactory.defaultMarker((it * 360 / numMarkersInRainbow)
+                        .toFloat()))
+                flat(flatBox.isChecked)
+                rotation(rotationBar.progress.toFloat())
+            })
+        }
+    }
+
+    /**
+     * Demonstrates converting a [Drawable] to a [BitmapDescriptor],
+     * for use as a marker icon.
+     */
+    private fun vectorToBitmap(@DrawableRes id : Int, @ColorInt color : Int): BitmapDescriptor {
+        val vectorDrawable: Drawable? = ResourcesCompat.getDrawable(resources, id, null)
+        if (vectorDrawable == null) {
+            Log.e(TAG, "Resource not found")
+            return BitmapDescriptorFactory.defaultMarker()
+        }
+        val bitmap = Bitmap.createBitmap(vectorDrawable.intrinsicWidth,
+                vectorDrawable.intrinsicHeight, Bitmap.Config.ARGB_8888)
+        val canvas = Canvas(bitmap)
+        vectorDrawable.setBounds(0, 0, canvas.width, canvas.height)
+        DrawableCompat.setTint(vectorDrawable, color)
+        vectorDrawable.draw(canvas)
+        return BitmapDescriptorFactory.fromBitmap(bitmap)
+    }
+
+    /**
+     * Checks if the map is ready, the executes the provided lambda function
+     *
+     * @param stuffToDo the code to be executed if the map is ready
+     */
+    private fun checkReadyThen(stuffToDo : () -> Unit) {
```

Comment:
Move these utility methods further down into one place.

Reply:
I have moved the checkReadyThen() method down to the bottom of the class. I'm not sure which other utility methods you mean?